src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -176,20 +175,27 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            DateTime now = DateTime.Now;
-            await InvokeJobFunction(now, false);
-
             // restart the timer with the next schedule occurrence
-            now = DateTime.Now;
-            DateTime nextOccurrence = _schedule.GetNextOccurrence(now);
-            TimeSpan nextInterval = nextOccurrence - now;
+            await InvokeJobFunction(DateTime.Now, false);
+            TimeSpan nextInterval = ScheduleStatus.Next - DateTime.Now;
```

Comment:
Is the change to use `DateTime.Now` (getting the time when this is calculated) instead of the cached value used when invoking the function intentional? This will behave differently as those could now be significantly different.